core/src/main/kotlin/com/atlassian/migration/datacenter/core/db/PostgresClientTooling.kt

```
@@ -66,18 +66,16 @@ class PostgresClientTooling(private val applicationConfiguration: ApplicationCon
     }
 
     /**
-     * Get the path of the pg_dump binary
+     * Get the path to the executable pg_dump binary
      *
      * @return the path to the dump utility
      */
     override fun getDatabaseDumpClientPath(): String? {
-        for (path in pddumpPaths) {
-            val p = Paths.get(path)
-            if (Files.isReadable(p) && Files.isExecutable(p)) {
-                return path
-            }
+        val pgDumpPath = resolvePgDumpPath()
+        return when(pgDumpPath != null && Files.isReadable(pgDumpPath) && Files.isExecutable(pgDumpPath)) {
+            true -> pgDumpPath.toString();
+            else -> null
```

Should we really be returning `null` here and forcing null checks everywhere? I think it would be more idiomatic to throw an exception as this is an error case i.e. there is no pg dump available

---

In this case  we dont want to throw an exception, `null` is used by the existing client of this refactored code to drive whether or not `pg_dump` is indicated as available or not on our initial trebuchet pre-req screen.

---

Ah I see, if this is the case then it's acceptable.